src/bp-members/bp-members-functions.php

```diff
@@ -5474,11 +5509,121 @@ function bb_set_bluk_user_profile_slug( $user_ids ) {
 	}
 
 	foreach ( $user_ids as $user_id ) {
-		$user_id           = (int) $user_id;
-		$unique_identifier = bb_generate_user_profile_slug( $user_id );
-		if ( ! empty( $unique_identifier ) ) {
-			bp_update_user_meta( $user_id, 'bb_profile_slug', $unique_identifier );
-			bp_update_user_meta( $user_id, 'bb_profile_slug_' . $unique_identifier, $user_id );
+		bb_set_user_profile_slug( (int) $user_id );
+	}
+}
+
+/**
+ * Function to generate the unique keys.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param int $max_ids How many unique IDs need to be generated? Default 1.
+ *
+ * @return array
+ */
+function bb_generate_uuids( $max_ids = 1 ) {
+	$max_ids         = absint( $max_ids );
+	$start           = 0;
+	$length          = 8;
+	$loop_count      = 1;
+	$max_length      = 12;
+	$generated_ids   = array(); // holds the generated ids.
+	$total_generated = 0;
+
+	while ( $total_generated < $max_ids ) {
+
+		$generated_ids[] = strtolower( substr( sha1( wp_generate_password( 12, false ) ), $start, $length ) );
+		while ( bb_is_exists_user_unique_identifier( end( $generated_ids ) ) ) {
```

Comment:
Instead of checking each ID's one by one we need to check them with single query. please check the flow again. if you need my help to understand, please coordinate with me. it's important for performance impact.